src/main/java/net/mcreator/ui/dialogs/ListEditorDialog.java

```diff
@@ -0,0 +1,133 @@
+
+	/**
+	 * Opens a dialog to edit string entries contained in the provided list.
+	 *
+	 * @param parent        The workspace window this method was called from.
+	 * @param textList      List of string entries that are about to be edited.
+	 * @param validator     Supplier of validators used on list entries' text fields, {@code null} means no validation.
+	 * @param uniqueEntries If {@code true}, duplicate list entries will not be allowed.
+	 * @return True if user chose OK option after editing strings list, false if it was cancel/close option.
+	 */
+	public static List<String> open(Window parent, List<String> textList,
+			@Nullable Function<VTextField, Validator> validator, boolean uniqueEntries) {
+		AtomicReference<List<String>> retVal = new AtomicReference<>();
+		MCreatorDialog dialog = new MCreatorDialog(parent, L10N.t("dialog.list_editor.title"), true);
+
+		List<ListEntry> entryList = new ArrayList<>();
+		JPanel entries = new JPanel(new GridLayout(0, 1, 5, 5));
+		for (String entry : textList)
+			new ListEntry(entryList, entries, entry, validator, uniqueEntries);
+
+		JButton add = new JButton(UIRES.get("16px.add.gif"));
+		add.setText(L10N.t("dialog.list_editor.add"));
+		add.addActionListener(e -> new ListEntry(entryList, entries, "", validator, uniqueEntries));
+
+		JPanel listPanel = new JPanel(new BorderLayout());
+		listPanel.add("North", PanelUtils.join(new JLabel(), add, new JLabel()));
+		listPanel.add("Center", new JScrollPane(PanelUtils.pullElementUp(entries)));
+
+		JButton ok = new JButton(UIManager.getString("OptionPane.okButtonText"));
+		JButton cancel = new JButton(UIManager.getString("OptionPane.cancelButtonText"));
+		dialog.getRootPane().setDefaultButton(ok);
+
+		ok.addActionListener(e -> {
+			if (new AggregatedValidationResult(
+					entryList.stream().map(s -> s.valueField).toArray(IValidable[]::new)).validateIsErrorFree()) {
+				retVal.set(new ArrayList<>());
+				for (ListEntry entry : entryList)
+					retVal.get().add(entry.valueField.getText());
+				dialog.setVisible(false);
+			} else {
+				JOptionPane.showMessageDialog(parent, L10N.t("dialog.list_editor.errors.message"),
+						L10N.t("dialog.list_editor.errors.title"), JOptionPane.ERROR_MESSAGE);
+			}
+		});
+		cancel.addActionListener(e -> dialog.setVisible(false));
+
+		dialog.getContentPane().add(PanelUtils.centerAndSouthElement(listPanel, PanelUtils.join(ok, cancel)));
+		dialog.setSize(420, 350);
+		dialog.setLocationRelativeTo(parent);
+		dialog.setVisible(true);
+
+		return retVal.get();
+	}
+
+	private static final class ListEntry extends JPanel {
+		private final VTextField valueField = new VTextField(20);
+
+		private ListEntry(List<ListEntry> entryList, JPanel parent, String value,
+				@Nullable Function<VTextField, Validator> validator, boolean uniqueEntries) {
+			super(new FlowLayout(FlowLayout.LEFT));
+			setOpaque(false);
+
+			Validator lev = validator == null ? null : validator.apply(valueField);

```

Comment:
`lev`?

Reply:
List Entry Validator (the `validator` name is already taken by a constructor parameter).